### PR TITLE
Fix t/test.pl and t/TEST to be strict compliant.

### PR DESCRIPTION
**Pull request by toddr**
So tests can run with -Dusedefaultstrict

### CODE REVIEW

File: t/test.pl [hunk 1]
```diff
@@ -1023,7 +1025,7 @@ sub _fresh_perl {
 
     # Use the first line of the program as a name if none was given
     unless( $name ) {
-        ($first_line, $name) = $prog =~ /^((.{1,50}).*)/;
+        my ($first_line, $name) = $prog =~ /^((.{1,50}).*)/;
```
Comment (Grinnz) [inline t/test.pl]: This doesn't look right, it shadows the outer $name instead of updating it.

Comment (Grinnz) [reply t/test.pl]: Probably should be `(my $first_line, $name) = ...`

File: t/test.pl [hunk 2]
```diff
@@ -1435,7 +1438,7 @@ ($@)
 sub new_ok {
     my($class, $args, $obj_name) = @_;
     $args ||= [];
-    $object_name = "The object" unless defined $obj_name;
+    my $object_name = "The object" unless defined $obj_name;
```
Comment (Grinnz) [inline t/test.pl]: Conditional my declaration, will lead to bugs.

File: t/test.pl (hunk 1 above)
Comment (toddr) [reply t/test.pl]: yep.

File: t/test.pl (hunk 2 above)
Comment (toddr) [reply t/test.pl]: actually this looks like a bug on closer inspection. `s/object_name/obj_name/g`

Comment (Grinnz) [reply t/test.pl]: I agree, it doesn't really make sense as written, this and the below usage likely should be using `$obj_name`.

Comment (toddr) [reply t/test.pl]: Fixed.

**Pull request merged**